lib/useKResponsiveWindow.js

```diff
@@ -0,0 +1,199 @@
+import { computed, onBeforeUnmount, onMounted, ref } from '@vue/composition-api';
+
+/**
+ * Returns window inner width and height
+ * @returns {Object} The window's inner width and height
+ */
+function windowMetrics() {
+  return {
+    width: window.innerWidth,
+    height: window.innerHeight,
+  };
+}
+
+/**
+ * Check if Nuxt is server side rendering
+ * @returns {Boolean}
+ */
+function isNuxtServerSideRendering() {
+  return process && process.server;
+}
+
+/**
+ * Add a media query
+ * @param {String} query
+ * @param {requestCallback} callback
+ * @returns {Object} The mediaQueryList, eventHandler, and stopListening
+ */
+function addMediaQuery(query, callback) {
+  if (isNuxtServerSideRendering()) {
+    return;
+  }
+
+  const mediaQueryList = window.matchMedia(query);
+  const eventHandler = event => {
+    callback(event);
+  };
+  const stopListening = () => {
+    mediaQueryList.removeEventListener('change', eventHandler);
+  };
+  mediaQueryList.addEventListener('change', eventHandler);
+
+  return {
+    mediaQueryList,
+    eventHandler,
+    stopListening,
+  };
+}
```

Comment:
This function is essentially a constructor. I think it's worth considering whether it would make sense to use a class here.

The benefits are fairly small for using a class here, but generally speaking they're more memory efficient. For instance, `stopListening` is a method for the returned object, and could take advantage of the classes prototype to avoid redeclaring it for every object. So whenever we define functions on returned objects, it's worth considering a class structure.

Perhaps it could also lead to some slight clarification, like for the `return;` when it's server side rendering. It isn't clear to me if this would end up raising errors in the server-side rendering since it returns an undefined value in this case. Perhaps that should be moved further up, if `useKResponsiveWindow` gets invoked in server-side rendering, leading to the invocation of `addMediaQuery`? Or is this check just a defensive measure and this code shouldn't be invoked?

Reply:
This is following the composable function style that VueJS suggests for using with the composition API: https://vuejs.org/guide/reusability/composables.html#what-is-a-composable

I've not seen uses of classes in this way, so there may be some gotchas around binding of methods when they are returned from the setup function in a Vue component.

Reply:
I don't think there are any gotchas. This composable format is an old JS style prior to the days of pure JS classes just repurposed for this. This function isn't a composable either!

Reply:
Ah sorry - yeah, this is just a utility function within the module. Carry on!

Reply:
Code has been refactored to factor this in. This file has been moved to the `lib/useKResponsiveWindow/index.js`